events_mgmt/src/test/java/com/adobe/event/management/ProviderServiceUpdateTestDrive.java

```diff
@@ -0,0 +1,116 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+package com.adobe.event.management;
+
+import com.adobe.Workspace;
+import com.adobe.event.management.model.EventMetadata;
+import com.adobe.event.management.model.Provider;
+import com.adobe.event.management.model.ProviderInputModel;
+import com.adobe.ims.JWTAuthInterceptor;
+import com.adobe.ims.util.PrivateKeyBuilder;
+import com.adobe.util.FileUtil;
+import feign.RequestInterceptor;
+import java.security.PrivateKey;
+import java.util.Optional;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProviderServiceUpdateTestDrive {
+
+  private static final Logger logger = LoggerFactory.getLogger(ProviderServiceUpdateTestDrive.class);
+
+  // use your own property file filePath or classpath and don't push back to git
+  private static final String DEFAULT_TEST_DRIVE_PROPERTIES = "workspace.secret.properties";
+  private static final String API_URL = "aio_api_url";
+
+  /**
+   * use your own property file filePath or classpath. WARNING: don't push back to github as it
+   * contains many secrets. We do provide a sample/template workspace.properties file in the
+   * ./src/test/resources folder
+   */
+  private static final String DEFAULT_TEST_PROPERTIES = "workspace.secret.properties";
+
+
+  public static void main(String[] args) {
+    try {
+
+      Properties prop =
+          FileUtil.readPropertiesFromClassPath(
+              (args != null && args.length > 0) ? args[0] : DEFAULT_TEST_DRIVE_PROPERTIES);
+
+      PrivateKey privateKey = new PrivateKeyBuilder().properties(prop).build();
+
+      Workspace workspace = Workspace.builder()
+          .properties(prop)
+          .privateKey(privateKey)
+          .build();
+
+      RequestInterceptor authInterceptor = JWTAuthInterceptor.builder()
+          .workspace(workspace)
+          .build();
+      
+      ProviderService providerService = ProviderService.builder()
+          .authInterceptor(authInterceptor) // [1]
+          .workspace(workspace) // [2]
+          .url(prop.getProperty(API_URL)) // you can omit this if you target prod
+          .build(); //
+
+      String instanceId = "testing_update_instance_id";
+      ProviderInputModel providerCreateModel = ProviderInputModel.builder()
+          .instanceId(instanceId)
+          .label("aio-lib-java test drive provider label")
+          .description("aio-lib-java test drive provider description")
+          .docsUrl(
+              "https://github.com/adobe/aio-lib-java/blob/main/events_mgmt/README.md#providerservice-test-drive")
+          .build();
```

Comment:
Where is the `provider_metadata` getting set in this input model? In the below code, we are assuming it is a 'custom-events' provider (https://github.com/adobe/aio-lib-java/pull/31/files#diff-e90890adf5de653ceb902a414cad5598b99496f98528314ec64561c1d48d3f62R89). Is the provider_metadata '3rd_party_custom_events' getting set somewhere in this case?

Reply:
when no `provider_metadata` is passed our API assumes we want to use the default `3rd_party_custom_events`